test/test_images/latencymako/sender_receiver.go

```diff
@@ -209,33 +228,29 @@ func (ex *senderReceiverExecutor) Run(ctx context.Context) {
 	printf("Starting events processor")
 	go ex.processEvents()
 
-	targeter := common.NewCloudEventsTargeter(sinkURL, msgSize, defaultEventType, eventsSource(), "binary").VegetaTargeter()
+	targeter := common.NewCloudEventsTargeter(sinkURL, msgSize, benchmarkEventType, eventsSource()).VegetaTargeter()
 
-	pacers := make([]vegeta.Pacer, len(ex.pacerSpecs))
-	durations := make([]time.Duration, len(ex.pacerSpecs))
+	attacks := make([]attackSpec, len(ex.pacerSpecs))
 	var totalBenchmarkDuration time.Duration = 0
 
 	for i, ps := range ex.pacerSpecs {
-		pacers[i] = vegeta.ConstantPacer{ps.rps, time.Second}
-		durations[i] = ps.duration
+		attacks[i] = attackSpec{pacer: vegeta.ConstantPacer{Freq: ps.rps, Per: time.Second}, duration: ps.duration}
 		printf("%d° pace: %d rps for %v seconds", i+1, ps.rps, ps.duration)
 		totalBenchmarkDuration = totalBenchmarkDuration + ps.duration
 	}
 
 	printf("Total benchmark duration: %v", totalBenchmarkDuration.Seconds())
 
-	combinedPacer, err := pkgpacers.NewCombined(pacers, durations)
-	if err != nil {
-		fatalf("Failed to setup combined pacer: %v", err)
-	}
-
 	printf("Starting benchmark")
 
-	vegetaResults := ex.attacker.Attack(targeter, combinedPacer, totalBenchmarkDuration, defaultEventType+"-attack")
-
 	// doneCh is closed as soon as all results have been processed
 	doneCh := make(chan struct{})
-	go ex.processVegetaResult(vegetaResults, doneCh)
+	go func() {
+		// Run all attacks
+		for i, f := range attacks {
+			f.Attack(i, targeter, ex.attacker)
+		}
+	}()
 	<-doneCh
```

Comment:
This will cause a deadlock because nothing closes that channel. Actually we probably don't need it anymore.